src/meta/src/hummock/manager/commit_epoch.rs

```diff
@@ -395,20 +398,24 @@ impl HummockManager {
                     })
                     .sum();
 
-                // TODO(li0k): replace with `split_sst`
-                let (modified_sst_info, branch_sst) = split_sst_with_table_ids(
-                    &sst.sst_info,
+                let sstable_info = sst.sst_info;
+                let (left, right) = split_sst(
+                    sstable_info,
                     &mut new_sst_id,
-                    origin_sst_size - new_sst_size,
+                    group_split::build_split_key(*match_ids.last().unwrap() + 1, VirtualNode::ZERO),
                     new_sst_size,
-                    match_ids,
+                    origin_sst_size - new_sst_size,
                 );
-                sst.sst_info = modified_sst_info;
 
-                commit_sstables
-                    .entry(group_id)
-                    .or_default()
-                    .push(branch_sst);
+                if let Some(left) = left {
```

Comment:
right, due to  line 382

```rust
if sst.sst_info.table_ids == match_ids {
                    // The SST contains all the tables in the group should be last key
                    assert!(index == len - 1);
                    commit_sstables
                        .entry(group_id)
                        .or_default()
                        .push(sst.sst_info);
                    break;
                }
```

I also think there should be Some on the right, otherwise it would break above